text-encode-transform.js

```diff
@@ -17,120 +17,113 @@
 (function() {
   'use strict';
 
-  if (TextEncoder.prototype.readable !== undefined &&
-      TextEncoder.prototype.writable !== undefined &&
-      TextDecoder.prototype.readable !== undefined &&
-      TextDecoder.prototype.writable !== undefined) {
-    return;
+  if (typeof self.TextEncoder !== 'function') {
+    throw new ReferenceError('TextEncoder implementation required');
```

Comment:
is it common that a library uses the ReferenceError like this, or could you please point me to some rationale (e.g. es spec section) supporting this use case?

Reply:
MDN describes it as "The ReferenceError object represents an error when a non-existent variable is referenced." which is close to the intent here.

The ES spec says "Indicate that an invalid reference value has been detected." which is not clear at all.

An alternative would be to log a console error and then do nothing. However, practically speaking this has the same effect, with the added benefit that if the developer hooks window.onerror they will get told about it.

Reply:
OK!